config/fi_provider.m4

```diff
@@ -192,3 +193,56 @@ AC_DEFUN([_FI_PROVIDER_INVOKE],[
 		[])]
 	    )
 ])
+
+dnl
+dnl Internal; should not be called from provder .m4 scripts.
+dnl Helper macro that appends each providers build flags for pkg-config
+dnl
+dnl Arguments:
+dnl
+dnl $1: name of the provider
+dnl
+AC_DEFUN([_FI_PROVIDER_PC_ADD],[
+	 FI_PC_CFLAGS="$FI_PC_CFLAGS $$1_CFLAGS $$1_CPPFLAGS"
+	 FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_LIBS"
+	 # usnic with nl needs special handling (should be added to usnic_LIBS?)
+	 AS_IF([test "$1" == "usnic"], [
+	  FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $$1_nl_LIBS"])
+])
+
+dnl
+dnl Substitute pkg-config variables after all providers have been initialized
+dnl If any package dependencies are ever needed, make use of
+dnl   FI_PC_REQUIRES_PKGS
+dnl   and
+dnl   FI_PC_PRIVATE_PKGS
+dnl
+AC_DEFUN([FI_PROVIDER_SETUP_PC],[
+	FI_PC_PRIVATE_LIBS="$FI_PC_PRIVATE_LIBS $LIBS"
+
+	dnl Try and avoid duplication of private library flags
+	dnl Inefficient but easier than tracking within each provider macro
+	seen=""
+	for arg in $FI_PC_PRIVATE_LIBS
+	do
+	  found=no
+	  for s in $seen
+	  do
+	    AS_IF([test "$s" = "$arg"], [found=yes])
+          done
+	  AS_IF([test "$found" = "no"], [seen="$seen $arg"])
+	done
+	FI_PC_PRIVATE_LIBS=$seen
+
+	AS_IF([test "x$enable_shared" != xyes],
```

Comment:
pkg-config does let you select; however, I've preferred to have dependent applications simply use `--libs` without `--static` in either case.  For example, with a shared and static build you might get:

```bash
$ PKG_CONFIG_PATH=. pkg-config --libs libfabric
-L/usr/local/lib -lfabric  
$ PKG_CONFIG_PATH=. pkg-config --static --libs libfabric
-L/usr/local/lib -lfabric -lpthread -ldl -lpsm_infinipath -lnl -lm  
```

With a static only build, you'd get:
```bash
$ PKG_CONFIG_PATH=. pkg-config --libs libfabric
-L/usr/local/lib -lfabric -lpthread -ldl -lpsm_infinipath -lnl -lm  
```

I find this to be a convenience but we can change if it bothers anyone.